docs/developer-guide/etendo-classic/concepts/Processes.md

```diff
@@ -0,0 +1,292 @@
+
+###  Executing processes
+
+!!!note
+    It is required to define the process compilation in order to be able to
+    execute it. This step can be performed executing ` ./gradlew smartbuild ` and
+    restarting Tomcat afterwards.  
+
+  
+Processes can be executed from the user interface from a menu or with a button. It can also be scheduled to be executed in the background without any user interaction.
+
+####  Executing a process from a menu option
+
+To execute a process from a menu option, you need to define a new menu option that executes the process. Menus are defined in `General Setup > Application > Menu`. In the *Action* field the *Process* entry must be selected, then in the *Process* field select the process. it will be executed immediately and then saved.
+
+Finally, press the *Tree* button to organize the new menu option in the menu tree, and now the process defined can be executed selecting this new menu created.
+
+####  Executing a process from a button
+
+To execute a process with a button you need to define a column that references a button. In `Application Dictionary > Tables and Columns`, go to the column you want to use to execute the process and in the field *Reference* select the *Button* entry, then in the *Process* field select the process you want to execute and save.
+
+When executing a process with a button, the record ID of the current record selected of the table will be passed to the process. This allows to execute functions for specific records.
+
+####  Executing a process in the background
+
+Backgroung processes are defined in `General Setup > Process Scheduling > Process Request`. 
+
+!!!info
+    For more information, read [How to create a Background Process](../../../developer-guide/etendo-classic/how-to-guides/How_to_create_a_Background_Process.md).
+
+In this window you can define a background process. The process to be executed can be selected in the *Process* field. Then you define the scheduling for the background process and it is then ready to use.
+
+!!!note
+    There is no user interaction necessary to execute the process, thus, no pop-up will appear prompting for additional parameters.
+
+#####  Kill a background Process
+ 
+  
+A button *Kill Process* is shown in the Process Monitor window on that processes that implement the KillableProcess interface while a process is being executed (Status = Processing). The interface KillableProcess will let you kill your process using a kill mechanism.
+
+**Mechanism to Kill**
+
+The mechanism that your background process uses to kill itself might vary between implementations. However, the main idea in any implementation should be to check some flag perodically during execution to see if a kill has been requested, and if the flag is set, somehow abort the execution of the rest of the job's work.
+
+
+Let's see an example,
+
+here we have a dummy process which simply prints the identifier in the log for all the business partners:
+
+    
```

Comment:
Dejo el código completo que saque del módulo que tiene los imports 
``` java title="DummyProcess.java"
package com.openbravo.example.killprocess.process;

import org.apache.log4j.Logger;
import org.hibernate.ScrollMode;
import org.hibernate.ScrollableResults;
import org.openbravo.dal.service.OBCriteria;
import org.openbravo.dal.service.OBDal;
import org.openbravo.model.common.businesspartner.BusinessPartner;
import org.openbravo.scheduling.ProcessBundle;
import org.openbravo.service.db.DalBaseProcess;
import org.openbravo.service.db.DbUtility;
import org.quartz.JobExecutionException;

public class DummyProcess extends DalBaseProcess {

  private static final Logger log4j = Logger.getLogger(DummyProcess.class);

  @Override
  protected void doExecute(ProcessBundle bundle) throws Exception {
    try {

      Thread.sleep(30000);
      // Get all business partners
      final OBCriteria<BusinessPartner> bpCri = OBDal.getInstance().createCriteria(
          BusinessPartner.class);
      bpCri.setFetchSize(1000);
      final ScrollableResults partnerScroller = bpCri.scroll(ScrollMode.FORWARD_ONLY);
      int i = 1;
      // Loop all business partners using a ScrollabeResults to avoid performance issues
      while (partnerScroller.next()) {
        final BusinessPartner bp = (BusinessPartner) partnerScroller.get()[0];
        // Print the Identifier for every business partner
        log4j.info(bp.getIdentifier());

        if ((i % 100) == 0) {
          OBDal.getInstance().getSession().clear();
        }
        i++;

      }
      partnerScroller.close();
    } catch (Exception ex) {
      Throwable e = DbUtility.getUnderlyingSQLException(ex);
      log4j.error("Error in DummyProcess", e);
      throw new JobExecutionException(e.getMessage(), e);
    }

  }

}

```